cli/command/stack/swarm/deploy_composefile.go

```diff
@@ -256,10 +264,29 @@ func deployServices(ctx context.Context, dockerCli command.Cli, services map[str
 				createOpts.QueryRegistry = true
 			}
 
-			if _, err := apiClient.ServiceCreate(ctx, serviceSpec, createOpts); err != nil {
-				return errors.Wrapf(err, "failed to create service %s", name)
+			response, err := apiClient.ServiceCreate(ctx, serviceSpec, createOpts)
+			if err != nil {
+				return errors.Wrapf(err, "Failed to create service %s", name)
 			}
+
+			serviceIDs = append(serviceIDs, response.ID)
 		}
 	}
-	return nil
+	return waitOnServices(ctx, dockerCli, serviceIDs, detach, quiet, errs)
+}
+
+func waitOnServices(ctx context.Context, dockerCli command.Cli, serviceIDs []string, detach bool, quiet bool, errs []string) error {
+	if !detach {
+		for _, serviceID := range serviceIDs {
+			if err := servicecli.WaitOnService(ctx, dockerCli, serviceID, quiet); err != nil {
+				errs = append(errs, fmt.Sprintf("%s: %v", serviceID, err))
```

Comment:
haven't tried if it'd look good, and probably fine for a follow-up, but we may be able to use the new `errors.Join` here, and create a multi-error 🤔

Reply:
@thaJeztah 👋 I have updated the code according to the suggestion by using `errors.Join()`.

Given the opportunity, I replaced `errors.Errorf` and `errors.Wrapf` with `fmt.Errorf` in this file, in order to avoid renaming the `errors` library.